### ISSUE TITLE
confusion about implementation of regression module during test time

### ISSUE THREAD

**Issue by magrenimish** [hqucv/siamban#56]
Hi,
I'm slightly confused as to how the 2 output channels of the classification map are used during test time and how the assignment of positive and negative labels for position (pi,pj) are identified during test time.
- from what I understand, the classification module returns 2 channels with the probability of the position (pi,pj) belonging to the foreground or the background. Since the classification module uses a cross-entropy loss, the ground-truth labels (1,0) are used for training.
- the ground-truth bounding box parameters are used to generate the ellipses (E1 and E2) and identify the positive and negative labels for each pixel and the regression is carried out only for (pi,pj) with a positive label
My question therefore is, how does the network identify the positive and negative labels to carry out regression during test time? Are the pixels with maximum foreground score considered to have positive labels?

**Comment by zeduchen** [comment]
Hi, in the inference stage, there is no ground truth, and the bounding box predicted by the position with the highest foreground score is used as the target position.

**Comment by magrenimish** [comment]
Thank you for the reply!
Similar to the RPN++ proposal selection strategy, do you also multiply the classification score with the temporal penalty and then perform Non Maximum Suppression?

**Comment by zeduchen** [comment]
> Thank you for the reply!
> Similar to the RPN++ proposal selection strategy, do you also multiply the classification score with the temporal penalty and then perform Non Maximum Suppression?

Similar to what you said.
